fdbserver/include/fdbserver/ProxyCommitData.actor.h

```diff
@@ -212,14 +212,18 @@ struct RangeLock {
 	}
 
 	void setPendingRequest(const Key& startKey, const RangeLockStateSet& lockSetState) {
-		ASSERT(SERVER_KNOBS->ENABLE_READ_LOCK_ON_RANGE);
+		ASSERT(SERVER_KNOBS->ENABLE_READ_LOCK_ON_RANGE && !SERVER_KNOBS->ENABLE_VERSION_VECTOR &&
```

Comment:
In general, I don't see a consistency in the checks that you are doing (for example, you didn't include a database lock check or an encryption check here). Maybe you can do refactoring here (include all these checks in a method)?

Reply:
Ditto